### DISCUSSION TITLE
Add option to make armor swapping work like bedrock edition.

### DISCUSSION THREAD

**Discussion by Tydium** [PurpurMC/Purpur#438] [__Completed__] [answered]
## Is your feature request related to a problem?
Kind of yes and no. 

## Describe the solution you'd like.
Basically my idea here is to add in the quick armor swapping from bedrock edition. I was going to think of putting this in a plugin instead, however it seems more suited to be as a purpur feature instead. The behavior that bedrock edition gives is that when you right click any armor piece it will automatically be put in the armor slot.

## Describe alternatives you've considered.
As said above I was originally going to look into it on a plugin. I spoke with a few devs and they said that it should be possible on the server side using plugins. (I figured server side would be better since theres alot of other people like me that want this feature.)

## Other
So I don't know much about coding, however there should be a packet for sending right clicks. So behavior should act like this:

If player sends "Right click" while holding "Armor piece" send to "armor slot"

(Example: If player sends "Right click" while holding "Netherite Chestplate" send to "Chestplate armor slot")

**Comment by emielderckx** [comment]
You can just shift click right. I dont really see the point

**Comment by encode42** [comment]
Discussed on Discord, feature already implemented in vanilla.

https://cdn.discordapp.com/attachments/685683802479001625/851517543042252820/2021-06-07_13-46-21.mp4

**Comment by encode42** [comment]
Re-opened because apparently a "non-gameplay modifying" performance or visual enhancement mod adds this ability.

**Comment by Tydium** [comment]
He was using client side mods. THIS IS NOT A VANILLA FEATURE

**Comment by Tydium** [comment]
So now basically if there is no armor in the slot using right click works, but if there is something in the slot already it does not work. So basically now what I am asking is to make it so when you right click while having armor already on is to switch out those pieces of armor.

**Comment by encode42** [comment]
For those wondering about the status of this issue, this patch is complete and will be merged once we are updated to 1.17.

**Comment by granny** [comment, marked as answer]
was implemented as of https://github.com/pl3xgaming/Purpur/commit/139f2f5d073c9f5ecf8186e14758f3bdf7cbed53
